packages/services/f-braze-adapter/src/BrazeAdapter.js

```diff
@@ -0,0 +1,55 @@
+import GetConsumerRegistry from './services/BrazeConsumerRegistry';
+import GetDispatcher from './BrazeDispatcher';
+
+/**
+ * sessionTimeoutInSeconds
+ * Set session timeout to 0 in order to avoid caching issues with Braze
+ * @type {number}
+ */
+export const sessionTimeoutInSeconds = 0;
+
+export default class BrazeAdapter {
+    /**
+     * Return the instance of the consumer that has been registered
+     * @returns {BrazeConsumer}
+     */
+    get consumerInstance () {
+        return this._consumer;
+    }
+
+    /**
+     * Return the dispatcher to perform card click events
+     * @returns {BrazeConsumer}
+     */
+    get dispatcher () {
+        return this._consumer;
+    }
+
+    constructor () {
+        // create the consumer registry
+        this._consumerRegistry = GetConsumerRegistry();
+        // create the dispatcher
+        this._dispatcher = GetDispatcher(sessionTimeoutInSeconds);
+    }
```

Comment:
IMO, for testability and maintainability the fewer singletons we're using, the better. As a relatively lightweight class, and the entry point to this module, I think this BrazeAdapter class would be better as the singleton, and that it should instantiate a non-singleton registry and dispatcher when it is constructed.

Reply:
Yes my thought process here was to have a BrazeAdapter class than can be instantiated to store the record of the consumer. The unsubscribe could be called on the instantiated BrazeAdapter object when no longer needed, it prevents the consumer of BrazeAdapter having to worry about the concept of consumers or registry. However I agree if we can figure out a way of having a less singletons then it would be better.

Reply:
One option would be to store a global variable for the consumers array on the `window` object, which would remove the need for the BrazeConsumerRegistry to be a singleton.

Reply:
Would another option would be to do `BrazeConsumerRegistry.prototype.consumers = [];` then consumers would be shared across instances?

Reply:
We could actually make all the methods static on the BrazeConsumerRegistry leaving us with one copy of `consumers` rather than a new one per instance?

Reply:
```
class BrazeConsumerRegistry {
    /**
     * Static constructor to store one instance of BrazeConsumerRegistry per js env
     * @return {BrazeConsumerRegistry}
     * @constructor
     */
    static consumers = [];

    /**
     * For each consumers content card callbacks fire the callback with supplied cards
     * @param cards
     */
    static applyContentCardCallbacks (cards) {
        this.consumers.reduce((acc, consumer) => [...acc, consumer.getContentCardCallbacks()])
            .forEach(callback => {
                callback(cards);
            });
    }

    /**
     * For each consumers in app message callbacks fire the callback with the supplied message
     * @param message
     */
    static applyInAppMessageCallbacks (message) {
        this.consumers.reduce((acc, consumer) => [...acc, consumer.getInAppMessagesCallbacks()])
            .forEach(callback => {
                callback(message);
            });
    }

    /**
     * For each consumers in app message click callbacks fire the callback with the supplied message
     * @param message
     */
    static applyInAppMessageClickEventsCallbacks (message) {
        this.consumers.reduce((acc, consumer) => [...acc, consumer.getInAppMessagesCallbacks()])
            .forEach(callback => {
                callback(message);
            });
    }

    /**
     * Register a consumer instance in the registry
     * @param consumerOptions
     * @returns {BrazeConsumer}
     */
    static register (consumerOptions) {
        const consumer = new BrazeConsumer(consumerOptions);
        this.consumers = [...this.consumers, consumer];
        return consumer;
    }

    /**
     * Remove the passed consumer from the registry
     * @param consumer {BrazeConsumer}
     */
    static unregister (consumer) {
        this.consumers = [...this.consumers.filter(c => c !== consumer)];
    }
}

export default BrazeConsumerRegistry;
```